proxy/InkAPI.cc

```diff
@@ -87,18 +88,28 @@
 static int api_rsb_index;
 static RecRawStatBlock *api_rsb;
 
-// Globals for the Sessions/Transaction index registry
-static int next_argv_index;
-
 static std::type_info const &TYPE_INFO_MGMT_INT   = typeid(MgmtInt);
 static std::type_info const &TYPE_INFO_MGMT_BYTE  = typeid(MgmtByte);
 static std::type_info const &TYPE_INFO_MGMT_FLOAT = typeid(MgmtFloat);
 
-static struct _STATE_ARG_TABLE {
-  char *name;
-  size_t name_len;
-  char *description;
-} state_arg_table[HTTP_SSN_TXN_MAX_USER_ARG];
+/** Reservation for a user arg.
+ */
+struct UserArg {
+  /// Types of user args.
+  enum Type {
+    TXN,  ///< Transaction based.
+    SSN,  ///< Session based
+    COUNT ///< Fake enum, # of valid entries.
+  };
+
+  std::string name;        ///< Name of reserving plugin.
```

Comment:
More of a question: we're sure we want to use std::string in the core? We made the decision to allow it?

Reply:
It's already there - see [here](https://github.com/apache/trafficserver/blob/master/proxy/http/HttpTransact.cc#L6409) for instance. I think it's clearly superior to `ats_scoped_str` so any place we use that and `ats_strdup` is better for being changed to `std::string`. In this case, I don't want to deal with string lifetime issues and ownership.